lumigator/python/mzai/schemas/__init__.py

```diff
@@ -0,0 +1,15 @@
+# Are these schemas used in both API and SDK?
+# As long as they're not used for other purposes
+# like DB storage, it should be ok
+
+# We could start with an OpenAPI spec and then generate the models
+# using https://docs.pydantic.dev/latest/integrations/datamodel_code_generator/
```

Comment:
TL;DR yes :) 

The idea is that since it's a remote invocation, and we're using python at both ends of the openapi, we should be using the same (or very similar) objects in both ends, the SDK and the server implementation. In that case there are never issues of mismatch between data sent and data received.

Usually one starts with an interface spec (CORBA, grpc, for example) and then derives code for the client and server (stubs and skeletons). In our case, we've specified the interface in pydantic, which allows us to walk in both directions: either from spec to code (the datamodel code generator) or from code to spec (model_to_json_schema or something like it). It's fine in both cases (as long as we provide the openapi spec in the end to external devs to be able to derive clients in other languages).

Sometimes the comms-related features (like serialization) in such data types are hidden behind an interface. In this case we derive from BaseModel, which is enough to provide serialization, and the SDK user will not need to know or use anything about it.